src/createFile.ts

```diff
@@ -70,10 +74,15 @@ export function createFileWithGeneratedTypes({ configFilename, outputFilename }:
     const allOpacities = extendedThemeOpacities ? { ...themeOpacities, ...extendedThemeOpacities } : themeOpacities;
     const opacities = Object.keys(allOpacities).map(opacity => `${prefix}opacity-${opacity}`);
 
-    const themeTextOpacities = isEmpty(THEME_CONFIG?.textOpacity) ? allOpacities : THEME_CONFIG?.textOpacity;
-    const extendedThemeTextOpacities = THEME_CONFIG?.extend?.textOpacity;
-    const allTextOpacities = extendedThemeTextOpacities ? { ...themeTextOpacities, ...extendedThemeTextOpacities } : themeTextOpacities;
-    const textOpacities = Object.keys(allTextOpacities).map(opacity => `${prefix}text-opacity-${opacity}`);
+    const getOpacity = (themePropertyName: string, outputNamePrefix: string) => {
+      const opacities = generateOpacities(allOpacities, THEME_CONFIG, themePropertyName);
```

Comment:
Looks good to me, great work, @dotellie ! 😃
do you think it's better to change this variable names to avoid shadowing?